src/snowflake/cli/_plugins/connection/util.py

```diff
@@ -54,6 +51,50 @@ def __init__(self, host: str | None):
         )
 
 
+class UIParameter(Enum):
+    NA_ENABLE_REGIONLESS_REDIRECT = "UI_SNOWSIGHT_ENABLE_REGIONLESS_REDIRECT"
+    NA_EVENT_SHARING_V2 = "ENABLE_EVENT_SHARING_V2_IN_THE_SAME_ACCOUNT"
+    NA_ENFORCE_MANDATORY_FILTERS = (
+        "ENFORCE_MANDATORY_FILTERS_FOR_SAME_ACCOUNT_INSTALLATION"
+    )
+
+
+def get_ui_parameter(
+    conn: SnowflakeConnection, parameter: UIParameter, default: Any
+) -> str:
+    """
+    Returns the value of a single UI parameter.
+    If the parameter is not found, the default value is returned.
+    """
+
+    ui_parameters = get_ui_parameters(conn)
+    return ui_parameters.get(parameter, default)
+
+
+@lru_cache(maxsize=None)
```

Comment:
LRU cache with no max size? Makes me a bit twitchy. In the context of a language server, for example, that could be a real issue. In any case this will be cached by connection, so it should be safe to use a rather small number here for the the LRU cache size.

Reply:
I removed maxsize, so I rely on the default, which is currently 128.